bin/lint-migrations-file/test/lint_migrations_file_test.clj

```diff
@@ -267,45 +288,66 @@
-                                    :rollback nil))))
+    (is-thrown-with-error-info?
+        {:error/message "Migration(s) ['v49.00-033'] uses invalid types (in 'boolean')"
+         :error/details {:invalid-ids ["v49.00-033"]
+                         :target-types #{"boolean"}}}
+      (validate (mock-change-set :id "v49.00-033"
+                                 :changes [{:modifyDataType {:newDataType "boolean"}}]
+                                 :rollback nil)))
 
-    (is (thrown-with-msg?
-         clojure.lang.ExceptionInfo
-         #"Validation failed"
-         (validate (mock-change-set :id "v45.12-345"
-                                    :changes [{:createTable {:columns [{:column {:type "boolean"}}]}}]
-                                    :rollback nil)))))
+    (is-thrown-with-error-info?
+        {:error/message "Migration(s) ['v49.00-033'] uses invalid types (in 'boolean')"
+         :error/details {:invalid-ids ["v49.00-033"]
+                         :target-types #{"boolean"}}}
+      (validate (mock-change-set :id "v49.00-033"
+                                 :changes [{:createTable {:tableName "my_table"
+                                                          :remarks "meow"
+                                                          :columns [{:column {:name "foo"
+                                                                              :remarks "none"
+                                                                              :type "boolean"}}]}}])))
+    (testing "does not throw for older migrations"
+      (is (validate (mock-change-set :id "v45.00-033"
+                                     :changes [{:createTable {:tableName "my_table"
+                                                              :remarks "meow"
+                                                              :columns [{:column {:name "foo"
+                                                                                  :remarks "none"
+                                                                                  :type "boolean"}}]}}])))))
 
   (testing "should throw if changes contains datetime type"
-    (is (thrown-with-msg?
-         clojure.lang.ExceptionInfo
-         #"Validation failed"
-         (validate (mock-change-set :id "v49.00-033"
-                                    :changes [{:modifyDataType {:newDataType "datetime"}}]
-                                    :rollback nil))))
+    (is-thrown-with-error-info?
+        {:error/message "Migration(s) ['v49.00-033'] uses invalid types (in 'timestamp','timestamp without time zone','datetime')"
+         :error/details {:invalid-ids ["v49.00-033"]
+                         :target-types #{"timestamp" "timestamp without time zone" "datetime"}}}
+      (validate (mock-change-set :id "v49.00-033"
+                                 :changes [{:modifyDataType {:newDataType "datetime"}}]
+                                 :rollback nil)))
 
-    (is (thrown-with-msg?
```

Comment:
This was passing for the wrong reason: validation failed because the `createTable` had missing required keys.

In fact, we don't want validation to fail in this case - it's an older ID, so we should allow the timestamp column.